src/app/components/App.tsx

```diff
@@ -1,109 +1,70 @@
-    StoredCourse | undefined
-  >();
-  const {
-    planners,
-    handleRemovePlanner,
-    handleAddPlanner,
-    handleSwitchPlanners,
-    handleChangePlannerName,
-  } = usePlanner(session?.user.id);
-
-  function handleShowCourseInfoModal(course: StoredCourse) {
-    setDisplayCourse(course);
-    setShowCourseInfoModal(true);
-  }
-
-  const Modals = () => (
-    <>
-      <CourseInfoModal
-        showModal={showCourseInfoModal}
-        setShowModal={setShowCourseInfoModal}
-        course={displayCourse}
-      />
-      <ExportModal
-        courseState={currentCourseState}
-        setShowModal={setShowExportModal}
-        showModal={showExportModal}
-      />
-      <MajorCompletionModal
-        setShowModal={setShowMajorCompletionModal}
-        showModal={showMajorCompletionModal}
-      />
-    </>
-  );
-
   return (
     <div className="h-full min-h-screen w-full bg-gray-100 flex flex-col justify-between">
       <ScreenSizeWarning />
-      {/* Header Start */}
-      <div className="">
-        <Navbar
-          setShowExportModal={setShowExportModal}
-          setShowMajorCompletionModal={setShowMajorCompletionModal}
-        />
-      </div>
-      {/* Header End */}
+      <ModalsStateProvider>
+        {/* Header Start */}
+        <Navbar />
+        {/* Header End */}
 
-      {/* Planner Start */}
-      <div className="pt-4 mb-auto">
-        <div className="flex justify-left px-7">
-          <TabList
-            planners={planners}
-            onRemovePlanner={handleRemovePlanner}
-            onAddPlanner={handleAddPlanner}
-            onSwitchPlanners={handleSwitchPlanners}
-            onChangePlannerName={handleChangePlannerName}
-          />
-        </div>
-        <div className="px-5">
-          <List>
-            {Object.keys(planners).map((id, index) => (
-              <ListItem
-                sx={{ display: planners[id][1] ? "block" : "none" }}
-                key={id}
-              >
-                <CoursePlanner
-                  order={index}
-                  title={planners[id][0]}
-                  id={id}
-                  isActive={planners[id][1]}
-                  onCourseStateChanged={setCurrentCourseState}
-                  onShowCourseInfoModal={handleShowCourseInfoModal}
-                />
-              </ListItem>
-            ))}
-          </List>
-        </div>
-        <Modals />
-      </div>
-      {/* Planner End */}
+        {/* Planner Start */}
+        <PlannersProvider>
+          <div className="pt-4 mb-auto">
+            <div className="flex justify-left px-7">
+              <TabList />
+            </div>
+            <div className="px-5">
+              <PlannerList />
+            </div>
+            <Modals />
+          </div>
+        </PlannersProvider>
+        {/* Planner End */}
+      </ModalsStateProvider>
 
       {/* Footer Start */}
       <Footer />
       {/* Footer End */}
     </div>
   );
 }
+
+const Modals = () => (
```

Comment:
Consistency issue: either make this a function or make PlannerList a variable assigned to an anonymous function